lib/srv/desktop/rdp/rdpclient/src/rdpdr/scard.rs

```diff
@@ -1320,6 +1482,25 @@ impl Status_Call {
     }
 }
 
+impl Encode for Status_Call {
+    fn encode(&self) -> RdpResult<Message> {
+        let mut w = vec![];
+
+        w.extend(RPCEStreamHeader::new().encode()?);
+        RPCETypeHeader::new(0).encode(&mut w)?;
+
+        let mut index = 0;
+        self.handle.encode_ptr(&mut index, &mut w)?;
+        let reader_names_is_null = if self.reader_names_is_null { 1 } else { 0 };
```

Comment:
You can also cast a bool value into an integer, e.g `self.reader_names_is_null as u32`. True will be 1 and false will be 0. 

You can read about it here: https://doc.rust-lang.org/std/primitive.bool.html